auth.go

```diff
@@ -426,69 +453,69 @@ func (x *Central) Logout(sessionkey string) error {
+	x.Log.Infof("SetPermit successful (%v)", userId)
+	return x.sessionDB.PermitChanged(userId, permit)
 }
 
 // Change a Password. This invalidates all sessions for this identity.
-func (x *Central) SetPassword(identity, password string) error {
-	if err := x.authenticator.SetPassword(identity, password); err != nil {
-		x.Log.Infof("SetPassword failed (%v) (%v)", identity, password)
+func (x *Central) SetPassword(userId UserId, password string) error {
+	if err := x.userstore.SetPassword(userId, password); err != nil {
+		x.Log.Infof("SetPassword failed (%v) (%v)", userId, password)
 		return err
 	}
-	x.Log.Infof("SetPassword successful (%v)", identity)
-	return x.sessionDB.InvalidateSessionsForIdentity(identity)
+	x.Log.Infof("SetPassword successful (%v)", userId)
+	return x.sessionDB.InvalidateSessionsForIdentity(userId)
 }
 
 // Create a one-time token that can be used to reset the password with a subsequent call to ResetPasswordFinish.
 // Any subsequent call to ResetPasswordStart causes the current token to be invalidated, so there can only
 // be a single active token. The token is valid until the time specified by 'expires'.
-func (x *Central) ResetPasswordStart(identity string, expires time.Time) (string, error) {
-	token, err := x.authenticator.ResetPasswordStart(identity, expires)
+func (x *Central) ResetPasswordStart(userId UserId, expires time.Time) (string, error) {
+	token, err := x.userstore.ResetPasswordStart(userId, expires)
 	if err != nil {
-		x.Log.Infof("ResetPasswordStart failed (%v) (%v)", identity, err)
+		x.Log.Infof("ResetPasswordStart failed (%v) (%v)", userId, err)
 		return "", err
 	}
-	x.Log.Infof("ResetPasswordStart successful (%v)", identity)
+	x.Log.Infof("ResetPasswordStart successful (%v)", userId)
 	return token, err
 }
 
 // Complete the password reset process, by providing a token that was generated by ResetPasswordStart.
 // If this succeeds, then the password is set to 'password', and the token becomes invalid.
-func (x *Central) ResetPasswordFinish(identity string, token string, password string) error {
-	if err := x.authenticator.ResetPasswordFinish(identity, token, password); err != nil {
-		x.Log.Infof("ResetPasswordFinish failed (%v) (%v)", identity, err)
+func (x *Central) ResetPasswordFinish(userId UserId, token string, password string) error {
+	if err := x.userstore.ResetPasswordFinish(userId, token, password); err != nil {
+		x.Log.Infof("ResetPasswordFinish failed (%v) (%v)", userId, err)
 		return err
 	}
-	x.Log.Infof("ResetPasswordFinish successful (%v)", identity)
-	x.sessionDB.InvalidateSessionsForIdentity(identity)
+	x.Log.Infof("ResetPasswordFinish successful (%v)", userId)
+	x.sessionDB.InvalidateSessionsForIdentity(userId)
 	return nil
 }
 
 // Create an identity in the Authenticator.
 // For the equivalent operation in the PermitDB, simply call SetPermit()
 func (x *Central) CreateAuthenticatorIdentity(identity, password string) error {
```

Comment:
This function is no longer accurately named. It should be called `CreateUserStoreIdentity`